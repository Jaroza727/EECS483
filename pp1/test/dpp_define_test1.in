#define EMPTY 
#EMPTY is empty

/* #define IN comment */
"#define IN string"
"# in string"
#IN

#define NAME name
"this is a good #NAME"
"this is a bad #name"
#NAMEisagoodname /* !! this one still requires clarification */

// redefine NAME
#define NAME new name
This is my #NAME
