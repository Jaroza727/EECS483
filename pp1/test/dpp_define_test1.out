
 is empty


"#define IN string"
"# in string"

*** Error line 7.
*** Invalid # directive




"this is a good #NAME"
"this is a bad #name"
nameisagoodname



This is my new name
